parallel_crawling_jobs.py

```diff
@@ -0,0 +1,58 @@
+# Launch parallel jobs in EC2 for crawling web pages
+import graphlab as gl
+import functools
+
+# Load Hacker News metadata SFrame from S3
+stories_sf = gl.load_sframe("s3://dato-datasets/hacker_news/stories.sframe")
+
+# Get a list of ID, URL pairs from SFrame
+id_url_pairs = [(x["id"], x["url"]) for x in stories_sf if x["url"]]
+
+# The base path to where source articles are to be stored in S3
+# Set this to a bucket to which you have write access
+s3_base_path = "s3://my-bucket/hacker_news/source_html"
+
+# Define a function to be called for each ID, URL pair
+@gl.deploy.required_packages(['requests == 2.3.0'])
+def get_source(id, url, s3_base_path, timeout=10):
+    import requests
+    import logging
+    import codecs
+    from graphlab.deploy._predictive_service._file_util import upload_to_s3
+
+    try:
+        response_txt = requests.get(url, timeout=timeout).text
+        filename = "%d.html" % id
+        with codecs.open(filename, "w", "utf8") as f:
+            f.write(response_txt)
+        s3_path = s3_base_path + "/" + filename
+        upload_to_s3(filename, s3_path)
+        return s3_path
+    except requests.exceptions.Timeout:
+        logging.getLogger().warning("Request for (%s, %d) timed out" % (url, id))
+    except Exception as e:
+        logging.getLogger().warning("Unexpected error on article (%s, %d): %s" % (url, id, e.message))
+
+# Divvy up the list of ID, URL pairs to distribute among 4 workers
```

Comment:
Did you actually want me to change "among" to "amongst"? "amongst" is less common in American English.